### PR TITLE
Cleanup zip and csv downloads and exports

### PR DESCRIPTION
**Pull request by knrafto**
Tested in staging:
* Exporting all Hog submissions
* Exporting scores for Hog
* Downloading my own submission for Maps

### CODE REVIEW

File: server/app/api.py [hunk 1]
```diff
@@ -1071,7 +1040,34 @@ def download(self, obj, user, data):
...
+            name, files = self.data_for_zip(obj)
+            write = start_response('200 OK', [
+                ('Content-Disposition', 'attachment; filename=submission-%s.zip' % name),
+                ('Content-Type', 'application/zip')
+            ])
+            stream = ResponseFile(write)
+            with zf.ZipFile(stream, 'w') as zipfile:
+                add_to_zip(zipfile, files)
+            return ''
+        return make_response(zip_response)
```
Comment (knrafto) [inline server/app/api.py]: I don't actually know that this streams the response instead of buffering it all in memory. Will investigate

**Comment by knrafto** [comment]
Now `-onlyfinal` works too :)

**Comment by Sumukh** [comment]
Cool - I'll take a look. I was under the impression that onlyfinal was working before?

**Comment by Sumukh** [comment]
Another super useful feature is something to proxy reads/writes to google storage. (aka let non-admins access files in the bucket. AFAIK this is possible by generating a signed url from the cloud storage api - but I'm not sure how it works.

**Comment by knrafto** [comment]
This cleanup was happening concurrently with the other PR, and I guess they diverged a bit.
I'll create an issue for the proxying

File: server/app/api.py (hunk 1 above)
Comment (knrafto) [inline server/app/api.py]: I think it streams: https://www.python.org/dev/peps/pep-0333/#the-write-callable is under the section "Buffering and Streaming"

**Comment by Sumukh** [comment]
Did you mean `-onlybackup`? I think all it needs is an index 
```
- kind: Backupv2
properties:
- name: assignment
- name: server_time
direction: desc
```

**Comment by Sumukh** [comment]
NVM. Just updated the indexes and downloading/searching by onlybackups works.

**Comment by knrafto** [comment]
So, a search query could return a FinalSubmission, a Submission, or a Backup, depending on what flags are used. Are there any others?

**Comment by knrafto** [comment]
`-onlybackup` works? This should still run into issue #604

**Comment by knrafto** [comment]
Unfortunately, downloading *all* Hog backups OOMs :( oh well

**Comment by Sumukh** [comment]
Closing due to master being on v3

**Pull request closed without merging**